deeprankcore/tools/transform.py

```diff
@@ -0,0 +1,199 @@
+
+            if node_features == "all":
+                node_features = available_node_features
+            if edge_features == "all":
+                edge_features = available_edge_features
+            if target_features == "all":
+                target_features = available_target_features
+
+            if not isinstance(node_features, list):
+                node_features = [node_features]
+            if not isinstance(edge_features, list):
+                edge_features = [edge_features]
+            if not isinstance(target_features, list):
+                target_features = [target_features]
+
+            # check node features
+            for feat in node_features:
+                if feat not in available_node_features:
+                    raise ValueError(
+                        f"The node feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable node features: {available_node_features}"
+                    )
+            # check edge features
+            for feat in edge_features:
+                if feat not in available_edge_features:
+                    raise ValueError(
+                        f"The edge feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable edge features: {available_edge_features}"
+                    )
+            # check target features
+            for feat in target_features:
+                if feat not in available_target_features:
+                    raise ValueError(
+                        f"The target feature _{feat}_ was not found in the file {hdf5_path}.\
+                        \nAvailable target features: {available_target_features}"
+                    )
+
+            df_dict = {}
+            df_dict['id'] = mol_keys
+
+            for feat in node_features + edge_features + target_features:
+                if feat in node_features:
+                    feat_type = 'node_features'
+                elif feat in edge_features:
+                    feat_type = 'edge_features'
+                else:
+                    feat_type = 'target_values'
+
+                if f[mol_key][feat_type][feat][()].ndim == 2:
+                    for i in range(f[mol_key][feat_type][feat][:].shape[1]):
+                        df_dict[feat + '_' + str(i)] = [f[mol_key][feat_type][feat][:][:,i] for mol_key in mol_keys]
+                else:
+                    df_dict[feat] = [
+                        f[mol_key][feat_type][feat][:]
+                        if f[mol_key][feat_type][feat][()].ndim == 1
+                        else f[mol_key][feat_type][feat][()] for mol_key in mol_keys]
+
+
+            df = pd.DataFrame(data=df_dict)
+
+        df_final = pd.concat([df_final, df])
+    
+    df_final.reset_index(drop=True, inplace=True)
+
+    return df_final
+
+
+def save_hist(
+    df: pd.DataFrame,
+    features: Union[str,List],
+    fname: str,
+    bins: Union[int,List,str] = 10,
+    figsize: Tuple = (15, 15)
```

Comment:
```suggestion
    df: pd.DataFrame,
    features: Union[str,List[str]],
    fname: str,
    bins: Union[int,List[float],str] = 10,
    figsize: Tuple = (15, 15)
```

Reply:
should we maybe default the bins to 'auto'?

Reply:
Often auto doesn't work well, especially for a huge number of continuous data points. 10 is the default used by Matplotlib, so I would leave that :)